Sources/Fluent/Query/Join.swift

```diff
@@ -18,10 +18,34 @@ public struct Join {
     /// the local data
     public let foreign: Entity.Type
```

Comment:
In case A pets is the child of users and in case B users is the child of pets. Maybe we have:

```swift
enum Join.Child {
    case foreign
    case local
}
```

?